### PR TITLE
Feature6

### PR DESCRIPTION
**Pull request by m05e5**
# Blog app - Add forms

- Create forms to perform the following functions:
    - Creates a Post on behalf of the current_user.
    - Create a comment on behalf of the current_user.

- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def create
+    @comment = Comment.new
+    # temp = params[:comment]
+    text = params[:text]
+    @comment.text = text
+    user = current_user
+    @comment.user_id = user.id
+    @comment.post_id = params[:post_id]
+
+    redirect_to(request.env['HTTP_REFERER']) if @comment.save
```
Comment (Davidosky007) [inline app/controllers/comments_controller.rb]: * Kindly add strong parameters here for example like this ▶️ `  
* `private `
   ` def post_params `
     ` params.require(:post).permit(:title, :text)`
   ` end`
* As This is a better security practice to help prevent accidentally allowing users to update sensitive model attributes.  [Refer back to lesson link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.)

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+    @post.comments_counter = 0
+    user = current_user
+    @post.user_id = user.id
+    temp = params[:post]
+    text = temp[:text]
+    title = temp[:title]
+    @post.text = text
+    @post.title = title
+
+    redirect_to("/users/#{user.id}/posts/#{@post.id}") if @post.save
```
Comment (Davidosky007) [inline app/controllers/posts_controller.rb]: * Kindly add strong parameters here for example like this ▶️ `  
* `private `
   ` def post_params `
     ` params.require(:post).permit(:title, :text)`
   ` end`
* As This is a better security practice to help prevent accidentally allowing users to update sensitive model attributes.  [Refer back to lesson link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @m05e5 👋, 

Good job so far! 💪 
* Good job using the right git-flow ✔️ 
* Properly written PR ✔️ 
* Descriptive commit messages ✔️ 
* No linter errors ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
Hi @m05e5 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada:


 #### Optional suggestions
 
 - N/A.
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**